lib/kafka_producer.js

```diff
@@ -55,6 +56,20 @@ function KafkaProducer(options, callback) {
         } else {
             self.statsd = false;
         }
+        if ('batching' in options) {
+            self.batching = options.batching;
+        } else {
+            self.batching = false;
+        }
+
+        if (self.batching) {
+            self.maxMessageSizeBytes = 1000100; // 1mb per a message
```

Comment:
Kafka does not accept messages larger than 1mb. If a message is larger than 1mb, it is dropped. This is used as a config.